### PR TITLE
add io.github.zhrexl.thisweekinmylife

### PR DESCRIPTION
**Pull request by zhrexl**
### Please confirm your submission meets all the criteria

- [x] I have read the [App Requirements][reqs] and [App Maintenance][maint] pages.
- [x] My pull request follows the instructions at [App Submission][submission].
- [x] I am using only the minimal set of permissions. *(If not, please explain each non-standard permission.)*
- [x] All assets referenced in the manifest are redistributable by any party.  If not, the unredistributable parts are using an extra-data source type.
- [x] I am an upstream contributor to the project. If not, I contacted upstream developers about submitting their software to Flathub. **Link:**
- [x] I own the domain used in the application ID or the domain has a policy for delegating subdomains (e.g. GitHub, SourceForge).
- [x] Any additional patches or files have been submitted to the upstream projects concerned. *(If not, explain why.)*

[reqs]: https://github.com/flathub/flathub/wiki/App-Requirements
[maint]: https://github.com/flathub/flathub/wiki/App-Maintenance
[submission]: https://github.com/flathub/flathub/wiki/App-Submission

### CODE REVIEW

**Comment by zhrexl** [comment]
bot, build com.gitub.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build com.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build com.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build com.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build com.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
I've no idea why it didnt accept the appdata.xml but i'm tired as hell and can't think straight right now

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by hfiguiere** [comment]
the build failed because no appstream file is found.

File: io.github.zhrexl.thisweekinmylife.json [hunk 1]
```diff
@@ -0,0 +1,41 @@
...
+    "modules" : [
+        {
+            "name" : "ThisWeekInMyLife",
+            "builddir" : true,
+            "buildsystem" : "meson",
+            "sources" : [
+                {
+                    "type" : "git",
+                    "url" : "https://github.com/zhrexl/ThisWeekInMyLife.git",
+                    "branch": "main",
```
Comment (hfiguiere) [inline io.github.zhrexl.thisweekinmylife.json:35]:
```suggestion
```

never put a branch.

File: io.github.zhrexl.thisweekinmylife.json [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+        "/share/gtk-doc",
+        "/share/man",
+        "/share/pkgconfig",
+        "*.la",
+        "*.a"
+    ],
+    "modules" : [
+        {
+            "name" : "ThisWeekInMyLife",
+            "builddir" : true,
```
Comment (hfiguiere) [inline io.github.zhrexl.thisweekinmylife.json:29]:
```suggestion
```

it's always true for meson

**Comment by zhrexl** [comment]
> the build failed because no appstream file is found.

Should I put the appdata or metainfo in this branch? Cause its already in the data folder and its supposed to be installed in the right directory

**Comment by hfiguiere** [comment]
it belong upstream, always.

**Comment by hfiguiere** [comment]
it has the wrong name

> Not exporting share/metainfo/io.github.zhrexl.ThisWeekInMyLife.metainfo.xml, non-allowed export filename

Same for the desktop file

**Comment by hfiguiere** [comment]
it's case sensitive.

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Comment by zhrexl** [comment]
> ```
> WARNING: Binary not found for Exec line in /srv/buildbot/worker/build-x86_64-3/build/builddir/export/share/applications/io.github.zhrexl.thisweekinmylife.desktop: ThisWeekInMyLife
> WARNING: Icon not matching app id in /srv/buildbot/worker/build-x86_64-3/build/builddir/export/share/applications/io.github.zhrexl.thisweekinmylife.desktop: io.github.zhrexl.ThisWeekInMyLife
> WARNING: Icon referenced in desktop file but not exported: io.github.zhrexl.ThisWeekInMyLife
> ```
> 
> It would have been so much easier to use the proper app-id here.

Yep. I messed up the files and as I don't know which command besides ```install``` to run ```flatpak-builder``` in order to reproduce the errors locally I'm using the bot to figure it out whats wrong.

**Comment by zhrexl** [comment]
bot, build io.github.zhrexl.thisweekinmylife

**Pull request closed without merging**